presto-product-tests/bin/run_on_docker.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash -ex
+
+# http://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-osx
+function absolutepath() {
+    [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+function retry() {
+  END=$(($(date +%s) + 600))
+
+  while (( $(date +%s) < $END )); do
+    set +e
+    "$@"
+    EXIT_CODE=$?
+    set -e
+
+    if [[ ${EXIT_CODE} == 0 ]]; then
+      break
+    fi
+    sleep 5
+  done
+
+  return ${EXIT_CODE}
+}
+
+function hadoop_master_container(){
+  docker ps --format '{{.Names}}' | grep hadoop-master
+}
+
+function check_hadoop() {
+  docker exec $(hadoop_master_container) supervisorctl status hive-server2 | grep -i running
+}
+
+function stop_unnecessary_hadoop_services() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl status
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop mapreduce-historyserver
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop yarn-resourcemanager
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop yarn-nodemanager
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop zookeeper
+}
+
+function docker_compose_network() {
+  docker network ls | grep ${ENVIRONMENT} | cut  -f 1 -d ' '
+}
+
+function check_presto() {
+  docker run \
+    --rm -it \
+    --net $(docker_compose_network) \
+    -v "${PROJECT_ROOT}/presto-cli/target/:/cli" \
+    teradatalabs/centos6-java8-oracle \
+    java -jar /cli/presto-cli-${PRESTO_VERSION}-executable.jar --server presto-master:8080 \
+    --execute "SHOW CATALOGS"  | grep -i hive
+}
+
+function run_product_tests() {
+  docker run \
+        --rm -it \
+        --net $(docker_compose_network) \
+        -v "${PRODUCT_TESTS_ROOT}:/presto-product-tests" \
+        teradatalabs/centos6-java8-oracle \
+        /presto-product-tests/bin/run.sh \
+        --config-local /presto-product-tests/conf/tempto/tempto-configuration.yaml "$@"
+}
+
+ENVIRONMENT=$1
+
+if [[ "$ENVIRONMENT" != "singlenode" && "$ENVIRONMENT" != "distributed" ]]; then
+   echo "Usage: run_on_docker.sh <singlenode|distributed> <product test args>"
+   exit 1
+fi
+
+shift 1
+
+SCRIPT_DIR=$(dirname $(absolutepath "$0"))
+PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
+PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
+DOCKER_COMPOSE_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/${ENVIRONMENT}/docker-compose.yml"
+
+# set presto version environment variable
+source "${PRODUCT_TESTS_ROOT}/target/classes/presto.env"
+
+# check docker and docker compose installation
+docker-compose version
+docker version
+
+# try to stop already running containers
+docker-compose -f "${PRODUCT_TESTS_ROOT}/conf/docker/singlenode/docker-compose.yml" down || true
```

Comment:
Stop just stops the container witout wiping out the data generated by the
tests. I might potentially cause the liveness issues when some of the tests
generates the data and doesn't remove it on finish. Down operation doesn't
remove the images, it only removes the container. So no redownloading is
being made.

On Thu, Mar 31, 2016, 13:00 Grzegorz Kokosiński <notifications@github.com>
wrote:

> In presto-product-tests/bin/run_on_docker.sh
> <https://github.com/prestodb/presto/pull/4877#discussion_r58035676>:
>
> > +shift 1
> > +
> > +SCRIPT_DIR=$(dirname $(absolutepath "$0"))
> > +PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
> > +PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
> > +DOCKER_COMPOSE_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/${ENVIRONMENT}/docker-
>
> compose.yml"
> > +
> > +# set presto version environment variable
> > +source "${PRODUCT_TESTS_ROOT}/target/classes/presto.env"
> > +
> > +# check docker and docker compose installation
> > +docker-compose version
> > +docker version
> > +
> > +# try to stop already running containers
> > +docker-compose -f "${PRODUCT_TESTS_ROOT}/conf/docker/singlenode/docker-compose.yml" down || true
>
> I would prefer to avoid using docker-compose down. In case this script is
> used locally then you need to download/build all the images every time. I
> think it is better to stop and then pull and build to make sure that images
> are fresh.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/prestodb/presto/pull/4877/files/c89ccaf31de9f8423f93b8b6a79b27a4b66b62d2#r58035676>
>